src/streaming-ingestion/flink-etl/src/main/java/software/aws/solution/clickstream/flink/StreamingJob.java

```diff
@@ -0,0 +1,419 @@
+//         appErrors.filter(value -> {
+//             LOG.error("Error: "+value);
+//             return false;
+//         });
+         matchedData.flatMap((String value, Collector<Tuple2<String, String>> out) -> {
+             try {
+                 JsonNode jsonNode = jsonParser.readValue(value, JsonNode.class);
+                 String compression = jsonNode.get("compression").asText();
+
+                 LOG.debug("data=" + jsonNode.get("data").asText());
+                 JsonNode dataNode = null;
+                 if ("gzip".equalsIgnoreCase(compression)) {
+                     GZIPInputStream gzip = null;
+                     BufferedReader br = null;
+                     StringBuilder output = new StringBuilder();
+                     try {
+                         byte[] decodedBytes = Base64.getDecoder().decode(jsonNode.get("data").asText().getBytes(StandardCharsets.UTF_8));
+                         gzip = new GZIPInputStream(new ByteArrayInputStream(decodedBytes));
+                         br = new BufferedReader(new InputStreamReader(gzip, StandardCharsets.UTF_8));
+
+                         String line;
+                         while ((line = br.readLine()) != null) {
+                             output.append(line);
+                         }
+                     } finally {
+                         if (br != null) br.close();
+                         if (gzip != null) gzip.close();
+                     }
+                     dataNode = jsonParser.readValue(output.toString(), JsonNode.class);
+                 } else {
+                     dataNode = jsonNode.get("data");
+                 }
+                 StringBuffer stringBuffer = new StringBuffer();
+                 stringBuffer.append("appId=").append(jsonNode.get("appId").asText())
+                         .append(",compression=").append(compression)
+                         .append(",is_array=").append(jsonParser.readTree(dataNode.toString()).isArray());
+                 LOG.info(stringBuffer.toString());
+                 ObjectNode ingestNode = jsonParser.createObjectNode();
+                 ingestNode.set("date", jsonNode.get("date"));
+//                 ingestNode.put("fakeIp", jsonNode.get("fakeIp").toString());
+                 ingestNode.put("ingest_time", jsonNode.get("ingest_time").asLong());
+                 ingestNode.set("ip", jsonNode.get("ip"));
+                 ingestNode.set("method", jsonNode.get("method"));
+                 ingestNode.set("path", jsonNode.get("path"));
+                 ingestNode.set("platform", jsonNode.get("platform"));
+                 ingestNode.set("rid", jsonNode.get("rid"));
+                 ingestNode.put("server_ingest_time", jsonNode.get("server_ingest_time").asLong());
+                 ingestNode.set("source_type", jsonNode.get("source_type"));
+                 ingestNode.set("timestamp", jsonNode.get("timestamp"));
```

Comment:
`asLong()`?

Reply:
The format is string, such as `2023-11-08T12:33:33.282166261Z`.